spectec/src/util/record.ml

```diff
@@ -0,0 +1,49 @@
+
+(* Helper functions for record *)
+module Record = struct include List
+  (* Record implements finite mapping *)
```

Comment:
Not sure how heavily this is used, but it seems rather inefficient. Wouldn't you rather want to use the OCaml lib's Map module for this purpose?

Reply:
That's because we wanted to keep the order of keys.

`Record` is mainly used for expressing the key-value structure in the spec such as
```
syntax funcinst hint(desc "function instance") =
  { TYPE deftype,
    MODULE moduleinst,
    CODE func }
```
and we wanted to keep this order when printing prose.

I must admit though, on second thought, there are some places where `Record` is used even if order of keys is not important. For those parts, we will refactor into using Map.

Reply:
I see, that makes sense. Yes, I was wondering about other uses, because the simple record use case wouldn't need most of the generic traversal functions in this module.